erts/doc/src/erlang.xml

```diff
@@ -11661,12 +11661,18 @@ improper_end</pre>
 
     <func>
       <name name="trunc" arity="1" since=""/>
-      <fsummary>Return an integer by truncating a number.</fsummary>
+      <fsummary>Truncates the decimals of  a number.</fsummary>
       <desc>
-        <p>Returns an integer by truncating <c><anno>Number</anno></c>,
+        <p>Truncates the decimals of <c><anno>Number</anno></c>,
           for example:</p>
         <pre>
-> <input>trunc(5.5).</input>
+> <input>trunc(5.7).</input>
+5</pre>
+        <pre>
+> <input>trunc(-5.7).</input>
+-5</pre>
+        <pre>
+> <input>trunc(5).</input>
```

Comment:
Maybe add an example of `trunc(36028797018963970.0)`? To illustrate that printed floats and integers may not work as expected.

Reply:
I have added it here: 516a316 . Maybe one should add an explanation as well but I'm not sure how to explain this in a good way. Feel free to comment if you have any suggestion?

Reply:
Maybe we need a separate chapter somewhere talking about the printing of floats? There are a bunch of places where this issue pops up, so having a central place to talk about it makes sense.

I tend to link to https://0.30000000000000004.com/ when explaining why things work as they do.

Reply:
That's a good site. I made the Elixir docs link to that and also https://floating-point-gui.de/

Reply:
The lack of representation of large integers as floats (and how they are printed) is not specific to trunc. So I think that should be explained some other place with maybe links from trunc and round.
For example:
trunc(36028797018963970.0) returns ..3968 because it got ..3968.0 as argument, not because of truncation.

Reply:
http://erlang.org/doc/reference_manual/data_types.html#number I think we should add a paragraph here that we link to in as many places as possible (trunc, round, float_to_list, io:format come to mind). That section should explain in brief why float printing works as it does and also link to some external sources for more detailed explanations.

Reply:
@garazdawi I think your suggestion is good. I will try to implement it.

Reply:
@garazdawi I have tried to implement your suggestion 002d1ef. Can you check it?